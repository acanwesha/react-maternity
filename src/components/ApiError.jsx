export const ApiError = props => {
    const { loading } = props
  
    if (loading) {
      return <div>My error.</div>
    }
}