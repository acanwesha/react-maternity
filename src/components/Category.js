import React, { useEffect, useState } from 'react'
import { Collapse, ListGroup, ListGroupItem } from "react-bootstrap"

const Category = props => {
  const {
    category,
    categoryIndex,
    selectedCategory,
    setSelectedCategory,
    methods,
    finalCategoryArray,
  } = props
  const { setCategoryArray } = methods

  const [collapsedState, setCollapsedState] = useState(
    categoryIndex === selectedCategory //  ===-1
  )
  const [subCheckbox, setSubCheckbox] = useState(false)

  useEffect(() => {
    setCollapsedState(categoryIndex === selectedCategory)
  }, [categoryIndex, selectedCategory])

  return (
    <div>
      <ListGroup>
        <span>
          <ListGroupItem style={{ backgroundColor: "#cceeff" }}>
            <div>
              <input
                type='checkbox'
                name='categoryId'
                data-name="category-checkbox"
                // id={category.id}
                onChange={e => {
                  setSubCheckbox(!subCheckbox)
                  const displayName = category.displayName
                  if (e.target.checked) {
                    //  if selected then added new category
                    const updatedArray = [
                      ...finalCategoryArray,
                      { displayName: displayName },
                    ]
                    setCategoryArray(updatedArray)
                    //                    console.log("added category", updatedArray)
                  } else {
                    //  if unselected then remove category from array
                    const updatedArray = finalCategoryArray.filter(
                      obj => obj.displayName !== displayName
                    )
                    setCategoryArray(updatedArray)
                    //                  console.log("removed category", updatedArray)
                  }
                }}
              />
              <label
                //    htmlFor={category.id}
                data-testid="displayName"
                style={{ cursor: "pointer" }}
                onClick={() => {
                  setSelectedCategory(categoryIndex)
                  setCollapsedState(!collapsedState)
                }}
                className='ml-2'
              >
                {category.displayName}
              </label>
            </div>
            <Collapse
              in={collapsedState}
              style={{
                position: "absolute",
                left: "348px",
                top: "0px",
                width: "450px",
                backgroundColor: "#dcf9fa"
              }}
            >
              <div id={category.id}>
                {category.result.map((item, index) => {
                  if (Object.keys(item).length === 0) {
                    return ""
                  }
                  //  if we unselect the main category then also need to unselect the sub categories
                  if (!subCheckbox) {
                    var checkboxElement = document.getElementById(item.id)
  
                    if (checkboxElement !== null) {
                      if (checkboxElement.checked === true) {
                        checkboxElement.checked = false
                      }
                    }
                  }
                  return (
                    <ListGroup 
                      key={index} 
                      className='ml-1'
                      data-name="item"
                    >
                      <span>
                        <ListGroupItem style={{ backgroundColor: "#caecfc" }}>
                          <div> 
                            <input
                              type='checkbox'
                              name='itemId'
                              data-name="sub-category-checkbox"
                              id={item.id}
                              disabled={!subCheckbox}
                              onChange={e => {
                                const displayName = item.displayName

                                //  first get the category index
                                const currentCategoryId =
                                  finalCategoryArray.findIndex(
                                    object =>
                                      object.displayName === category.displayName
                                  )
                                //  create a duplicate array for modification
                                const duplicateArray = [...finalCategoryArray]
                                var resultArray = []
                                if (
                                  duplicateArray[currentCategoryId].result !==
                                  undefined
                                ) {
                                  resultArray = [
                                    ...duplicateArray[currentCategoryId].result,
                                  ]
                                }
                                if (e.target.checked) {
                                  //  added into array
                                  const resultObject = {
                                    displayName: displayName,
                                  }
                                  if (item.value !== undefined) {
                                    resultObject["value"] = item.value
                                  }
                                  resultArray.push(resultObject)
                                  duplicateArray[currentCategoryId].result =
                                    resultArray
                                  setCategoryArray(duplicateArray)
                                   console.log(
                                    "sub-category added ",
                                    duplicateArray
                                  )
                                } else {
                                  // removed from array
                                  duplicateArray[currentCategoryId].result =
                                    resultArray.filter(
                                      object => object.displayName !== item.displayName
                                    )

                                  setCategoryArray(duplicateArray)
                                  console.log(
                                    "sub-category removed ",
                                    duplicateArray
                                  )
                                }
                              }}
                            />
                            <label
                              htmlFor={item.id}
                              style={{ cursor: "pointer" }}
                              className='ml-2'
                            >
                              {item.displayName}
                            </label>
                            {item.value !== undefined && (
                              <span>
                                : <b> {item.value}</b>
                              </span>
                            )}
                          </div>
                        </ListGroupItem>
                      </span>
                    </ListGroup>
                  )
                })}
              </div>
            </Collapse>
          </ListGroupItem>
        </span>
      </ListGroup>
    </div>
  )
}
export default Category;
