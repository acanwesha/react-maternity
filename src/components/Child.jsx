import { Card } from "react-bootstrap"

export const Child = props => {
  const { loading, child } = props

  if (loading) {
    return <div>loading......</div>
  }

  return (
    <div>
      <div>
        <h4 className='text-center'>Child</h4>
        <Card className='my-2'>
          <Card.Body style={{ backgroundColor: "#cceeff" }}>
            <Card.Text className='font-weight-bold'>
              Name : &nbsp;
              <span className='font-weight-light'>
                {child.firstName + " " + child.lastName}
              </span>
            </Card.Text>
            <Card.Text className='font-weight-bold'>
              MRN : &nbsp;<span className='font-weight-light'>{child.mrn}</span>
            </Card.Text>
            <Card.Text className='font-weight-bold'>
              FIN : &nbsp;<span className='font-weight-light'>{child.fin}</span>
            </Card.Text>
          </Card.Body>
        </Card>
      </div>
    </div>
  )
}
