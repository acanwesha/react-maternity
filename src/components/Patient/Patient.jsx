import { Card } from "react-bootstrap"

export const Patient = props => {
  const { loading, patient } = props

  if (loading) {
    return <div>loading......</div>
  }

 return (
    <div>
      <div>
        <h4 className='text-center'>Patient</h4>
        <Card className='my-2'>
          <Card.Body style={{ backgroundColor: "#cceeff" }}>
            <Card.Text className='font-weight-bold'>
              Name : &nbsp;
              <span data-testid="name" className='font-weight-light'>
                {patient.firstName + " " + patient.lastName}
              </span>
            </Card.Text>
            <Card.Text className='font-weight-bold'>
              MRN : &nbsp;<span data-testid="mrn" className='font-weight-light'>{patient.mrn}</span>
            </Card.Text>
            <Card.Text className='font-weight-bold'>
              FIN : &nbsp; <span data-testid="fin" className='font-weight-light'>{patient.fin}</span>
            </Card.Text>
          </Card.Body>
        </Card>
      </div>
    </div>
  )
}
