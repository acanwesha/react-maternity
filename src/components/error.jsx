import { Jumbotron } from "react-bootstrap"
const Error = () => {
  return (
    <div>
      <Jumbotron>
        <h1>404 - Not Found!</h1>
      </Jumbotron>
    </div>
  )
}

export default Error
