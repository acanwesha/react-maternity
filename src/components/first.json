{
  "patient": [
    {
      "patientDetails": {
        "id": 2,
        "firstName": "SOPHIA",
        "lastName": "ROBERTS",
        "mrn": "MH36549879",
        "fin": "MH46846589"
      }
    }
  ],
  "child": [
    {
      "childDetails": [
        {
          "resultCopiedDateTime": "2021-06-06T18:30:00",
          "id": 6,
          "firstName": "AALIYAH",
          "lastName": "ROBERTS",
          "mrn": "MH112973 ",
          "fin": "MH284550"
        }
      ]
    }
  ],
  "category": []
}
