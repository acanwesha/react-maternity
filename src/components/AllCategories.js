import React, { useState } from "react"
import Category from "./Category"
import axios from 'axios';

const AllCategories = props => {
  const { data, methods, finalCategoryArray } = props
  const [selectedCategory, setSelectedCategory] = useState(-1)
  console.log("data", data)
  return (
    <div>
      <h3>Category</h3>
      <p></p>
      {data.length > 0
        ? data.map((item, index_1) => {
            return (
              <Category
                key={index_1}
                category={item}
                categoryIndex={index_1}
                finalCategoryArray={finalCategoryArray}
                selectedCategory={selectedCategory}
                setSelectedCategory={setSelectedCategory}
                methods={methods}
              />
            )
          })
        : "No categories available"}
    </div>
  )
}
export default AllCategories
