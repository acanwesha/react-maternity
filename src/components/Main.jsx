import { Container, Row, Col, Dropdown, Button, Alert, Jumbotron } from "react-bootstrap"
import { Patient } from "./Patient/index"
import Category from "./Category"
import AllCategories from "./AllCategories"
import React, { useRef } from "react"
import { Child } from "./Child"
import second from "./second.json"
import axios from "axios"
import { ApiError} from "./ApiError"
export class Main extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      loading: true,
      data: {},
      patientId: props.match.params.id,
      childId: 0,
      finalCategory: [],
      condition: false,
      status: false,
      count: 1,
    }
  }
  setFinalCategoryArray = object => {
    this.setState({ finalCategory: object })
  }
  getRequest(id) {
    //  need to take id from url
    /*
    if (/^\d*[1-9]\d*$/.$id){
      return (
        <div>
          <Jumbotron> 
            <h1>Enetr a positive value</h1>
          </Jumbotron>
        </div>
      )
    }*/
    
    axios
      .get(`http://localhost:8082/result-copy-war/api/patient-results/${id}`)
      .then(response => {
        this.setState({ data: response.data, loading: false })
        console.log(response)
        //
        //
      })
      .catch(err => console.log("getRequest api error ", err))
  }
  //  post body that is sent 
  async postRequest(body) {
    console.log("body", body)
    axios
      .post(`http://localhost:8082/result-copy-war/api/copy-results`, body,
      {
        mode: "no-cors",
        method: "POST",
        headers: {
          "Accept": "application/json",
          "Content": "application/json",
          "Access-Control-Allow-Origin" : "*",
          "Access-Control-Allow-Methods" : "POST,GET,PUT,DELETE",
          "Access-Control-Allow-Headers" : "Authorization, Lang, Content-Type"  , 
        }
      }
      )
      .then(response => {
        console.log("postRequest api response", response)
      })
      .catch(error => console.log("postRequest api error ", error))
  }

  componentDidMount() {
    if (true) {
      this.getRequest(this.state.patientId)
    } else {
      setTimeout(() => {
        this.setState({ data: second, loading: false })
      }, 3000)
    }
  }
  setChildId(childId) {
    this.setState({ childId: childId })
  }
  render() {
    //  this is for patient
    var patient = {}
    if (!this.state.loading) {
      patient = this.state.data.patient[0].patientDetails
    }
    //  this is for child
    var child = {}
    if (!this.state.loading) {
      child = this.state.data.child[0].childDetails[this.state.childId]
    }

    //  this is for total childrens
    var childrenArray = []
    if (!this.state.loading) {
      childrenArray = this.state.data.child[0].childDetails
    }
    
    //  this is for category array
    var categoryArray = []
    if (!this.state.loading) {
      categoryArray = this.state.data.category
    }
    const methods = {
      setCategoryArray: this.setFinalCategoryArray,
    }

    return (
    
      <Container>
        <Row>
          <Col>
            <Patient loading={this.state.loading} patient={patient} />
          </Col>
          <Col>
            <Child loading={this.state.loading} child={child} />
          </Col>
        </Row>
        <Row className='my-3'>
          <Col>
            {  child.resultCopiedDateTime != undefined 
            ? (
              
              <Alert variant='primary' className='text-center'>
                Result was copied on :  
               <b> {child.resultCopiedDateTime.split("T")[0]} </b>
               for  {child.firstName} {child.lastName}  
              </Alert>
            ) : (
              
              <div></div>
            )}
          </Col>
        </Row>
        <Row>
          {childrenArray.map((object,index) => {
            if(object.resultCopiedDateTime === undefined)
            {
              this.state.status = false
              return ""
            }
             this.state.status = true//this.state.status + 1
          })
          }
          {this.state.status ? (

               <Alert variant='primary' className='text-center' style={{ marginLeft:"16px" , marginTop:"-35px"
               , borderBlockStyle:"none", width:"1110px", height:"40px"}}
               >Results for all the children are copied
               </Alert>
          ):(
                ""
          )} 
        </Row>
        <Row>
          <Col>
          {child.resultCopiedDateTime === undefined || 
            childrenArray.length > 1
             ? (
              <Dropdown>
              <Dropdown.Toggle variant='success' id='dropdown-basic'>
                Select Child
              </Dropdown.Toggle>
              {childrenArray.length > 0 ? (
                <Dropdown.Menu>
                  {childrenArray.map((obj, index) => {
                    return (
                      <Dropdown.Item
                        key={index}
                        onClick={() => {
                          this.setChildId(index)
                        }}
                      >
                        Baby {String.fromCharCode(index + 65)}
                      </Dropdown.Item>
                    )
                  })}
                </Dropdown.Menu>
              ) : (
                ""
              )}
            </Dropdown>
            ) : (
              <div></div>
            )}
            
          </Col>
        </Row>
        <Row className='my-1'>
          <Col md={4}>
            {!this.state.loading &&
              child.resultCopiedDateTime === undefined && (
                <AllCategories
                  data={categoryArray}
                  finalCategoryArray={this.state.finalCategory}
                  methods={methods}
                />
              )}
          </Col>
        </Row>
        <Row className='my-1 text-center'>
          <Col>
            {child.resultCopiedDateTime === undefined 
            ? (
                <Button
                className='ml-1'
                onClick={() => {
                  if (!this.state.loading) {
                    if (child.resultCopiedDateTime === undefined) {
                      if (this.state.finalCategory.length === 0) {
                        alert("Select category!.")
                      } else {
                        var body = {
                          childId: child.id,
                          category: this.state.finalCategory,
                        }
  
                        alert(JSON.stringify(body))
                        this.postRequest(body)
                      }
                    } 
                  }
                }}
                variant='primary'
                style={{ marginTop: "120px" }}
              >
                Result Copy
              </Button>
              ) : (
                <div></div>
              )}
          </Col>
        </Row>
        <div style={{ marginBottom: "200px" }} />
      </Container>
    )
  }
}

export default Main

