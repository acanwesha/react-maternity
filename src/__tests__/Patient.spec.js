import '@testing-library/jest-dom'
import * as React from 'react'
import renderer from 'react-test-renderer';
import { Patient } from '../components/Patient';
import { render, screen } from '@testing-library/react'
const patient = {
  "firstName": "EMMA",
  "lastName": "ESPINOSA",
  "mrn": "test",
  "fin": "test 2u"
}

describe('Patient', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Patient patient={patient} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("show props correctly", () => {
    render(<Patient patient={patient} />)
    expect(screen.getByTestId('name')).toHaveTextContent(`${patient.firstName} ${patient.lastName}`)
    expect(screen.getByTestId('mrn')).toHaveTextContent(patient.mrn)
    expect(screen.getByTestId('fin')).toHaveTextContent(patient.fin)
  })
})