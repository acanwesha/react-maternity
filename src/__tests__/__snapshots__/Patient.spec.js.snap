// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient renders correctly 1`] = `
<div>
  <div>
    <h4
      className="text-center"
    >
      Patient
    </h4>
    <div
      className="my-2 card"
    >
      <div
        className="card-body"
        style={
          Object {
            "backgroundColor": "#cceeff",
          }
        }
      >
        <p
          className="font-weight-bold card-text"
        >
          Name :  
          <span
            className="font-weight-light"
            data-testid="name"
          >
            EMMA ESPINOSA
          </span>
        </p>
        <p
          className="font-weight-bold card-text"
        >
          MRN :  
          <span
            className="font-weight-light"
            data-testid="mrn"
          >
            test
          </span>
        </p>
        <p
          className="font-weight-bold card-text"
        >
          FIN :   
          <span
            className="font-weight-light"
            data-testid="fin"
          >
            test 2u
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
